### ISSUE TITLE
Could not find a declaration file for module 'eslint-plugin-import'

### ISSUE THREAD

**Issue by fabulousgk** [import-js/eslint-plugin-import#3090]
Using latest eslint with flat config in a TS project.  Need type declarations.

**Comment by ljharb** [comment]
We don't currently ship any (and they're not required to be able to use the plugin)

**Comment by soryy708** [comment]
Try this:

```
npm install eslint-plugin-import --save-dev
```

as part of https://github.com/import-js/eslint-plugin-import?tab=readme-ov-file#installation

**Comment by fabulousgk** [comment]
Yeah, so the issue it that typescript-eslint tells you to put //@ts-check on your config file which then produces a slew of errors due to missing type declarations.  To be fair this is not the only package that is missing type defs, so yeah.  For now I am just removing that, really no need to be on the config tbh.

Would still be nice to have them, but probably not a priority.

**Comment by Sanshain** [comment]
have the same type checking problem, but the plugin itself works

**Comment by appsparkler** [comment]
```ts
// ./eslint-plugin-import.d.ts
declare module "eslint-plugin-import" {}
```

**Comment by basselalsayed** [comment]
Adding on to @appsparkler's solution, as I was getting a typescript-eslint error about unsafe access due to lack of types:

```
declare module 'eslint-plugin-import' {
  import type { Linter } from 'eslint';

  const importPlugin: {
    flatConfigs: {
      recommended: Linter.Config;
      typescript: Linter.Config;
    };
  };

  export default importPlugin;
}
```

**Comment by JounQin** [comment]
@basselalsayed Or you can just install this plugin as npm dep via github protocol.

**Comment by basselalsayed** [comment]
> [@basselalsayed](https://github.com/basselalsayed) Or you can just install this plugin as npm dep via github protocol.

I have the latest version installed in my package.json but the types didn't come through. I can see the package has some [here](https://github.com/import-js/eslint-plugin-import/blob/main/index.d.ts)

**Comment by JounQin** [comment]
> I have the latest version installed in my package.json but the types didn't come through. I can see the package has some [here](https://github.com/import-js/eslint-plugin-import/blob/main/index.d.ts?rgh-link-date=2025-05-01T15%3A13%3A00.000Z)

Please read more carefully, not via npm but:

> install this plugin as npm dep via github protocol.

https://docs.npmjs.com/cli/v11/configuring-npm/package-json#github-urls

**Comment by basselalsayed** [comment]
> > I have the latest version installed in my package.json but the types didn't come through. I can see the package has some [here](https://github.com/import-js/eslint-plugin-import/blob/main/index.d.ts?rgh-link-date=2025-05-01T15%3A13%3A00.000Z)
> 
> Please read more carefully, not via npm but:
> 
> > install this plugin as npm dep via github protocol.
> 
> https://docs.npmjs.com/cli/v11/configuring-npm/package-json#github-urls

That shouldn't make a difference and it breaks updates iirc. Are you a maintainer? If so, thank you it's a great project, but can't you add a "types" key in the package.json? https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

**Comment by basselalsayed** [comment]
> > I have the latest version installed in my package.json but the types didn't come through. I can see the package has some [here](https://github.com/import-js/eslint-plugin-import/blob/main/index.d.ts?rgh-link-date=2025-05-01T15%3A13%3A00.000Z)
> 
> Please read more carefully, not via npm but:
> 
> > install this plugin as npm dep via github protocol.
> 
> https://docs.npmjs.com/cli/v11/configuring-npm/package-json#github-urls

Why does this make a difference? It's more hassle tbh

**Comment by JounQin** [comment]
> Why does this make a difference? It's more hassle tbh

This is how it works as the owber, @ljharb, prefers. And it's not hassle at all IMO, just change one version into another `import-js/import-plugin-import`, how is that more hassle?

**Comment by ljharb** [comment]
It’s not a “preferred” mechanism - things shouldn’t be installed from git - it’s just that the types change isn’t released yet.

**Comment by JounQin** [comment]
> It’s not a “preferred” mechanism - things shouldn’t be installed from git - it’s just that the types change isn’t released yet.

The issue is exactly that the types change isn’t released yet. 😂

**Comment by basselalsayed** [comment]
> > Why does this make a difference? It's more hassle tbh
> 
> This is how it works as the owber, [@ljharb](https://github.com/ljharb), prefers. And it's not hassle at all IMO, just change one version into another `import-js/import-plugin-import`, how is that more hassle?

It's more hassle because what I have is working fine for me now 😂.

Changing the management for one dependency that goes against the rest of the ecosystem adds mental overhead. It might affect my dependabot, or my regular `pnpm update`. Perhaps not, but I'm unfamiliar with this unless there is an urgent bug fix or an unpublished fork that had to be installed directly from git.

Glad to see it is a scheduled update. Choosing git as the delivery mechanism is scary. Thanks for all your work on this project.

**Comment by JounQin** [comment]
Or maybe just change to use another library... 😂
